### DISCUSSION TITLE
Stream upload file for pre-recorded processing.

### DISCUSSION THREAD

**Discussion by ggratte** [deepgram/community#676] [Feature Requests]
I want to process the recording in one chunk but start streaming as I record it, so the MP3 is completely uploaded and ready for processing once I'm finished speaking. To reduce the uploading time. Is this possible?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @ggratte, our two transcription options are streaming and pre-recorded. We don't support streaming audio to us for pre-recorded transcription.

Have you tried out sending Deepgram a file and found that it has significant latency? If so, how long is your audio? One option that should have low latency is sending us a URL to a cloud audio file, so the file doesn't have to upload from the client side.

**Reply by ggratte** [reply]
It takes about 2 seconds for a 1-minute recording request, which is quite much for a responsive UI. I tried the streaming first but recognized that the pre-recorded output is significantly better for the recordings I've tested.
I guess uploading the mp3 is the bulk of that request time? I like your idea about streaming to the cloud and sending you the URL!
